### DISCUSSION TITLE
Ollama Support?

### DISCUSSION THREAD

**Discussion by mrjones2014** [olimorris/codecompanion.nvim#7] [Ideas]
I've recently been using [Ollama](https://github.com/ollama/ollama), which basically provides a unified REST API interface to any open source LLM model, with [Open-WebUI](https://github.com/open-webui/open-webui), which basically gives a completely offline local ChatGPT-like experience.

Any chance we could make like an API adapter or something to make this plugin work with a local Ollama API?

**Comment by mrjones2014** [comment]
Even better would be to use the Open-WebUI API directly, as that project adds some features like chat history etc.

**Comment by olimorris** [comment]
Great timing. I only came across Ollama a couple of days ago. Looks like integration would be pretty straight forward given the API structure but yep, will need adapters. The code base is _verrrrryyyyyyyy_ tightly coupled to OpenAI.

**Reply by mrjones2014** [reply]
Amazing, maybe I can find some time this weekend to really dig into the code to help out.

**Reply by olimorris** [reply]
I've made a first stab at what an adapter would look like in a PR.

**Comment by mrjones2014** [comment]
Hey @olimorris I'm trying this out and I'm getting this error:

```
CodeCompanion Error executing Lua callback: .../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: Vim:Error executing Lua callback: ...decompanion.nvim/lua/codecompanion/strategies/inline.lua:224: attempt to call method 'set_params' (a nil value)
stack traceback:
	...decompanion.nvim/lua/codecompanion/strategies/inline.lua:224: in function 'inline'
	/home/mat/git/codecompanion.nvim/plugin/codecompanion.lua:12: in function </home/mat/git/codecompanion.nvim/plugin/codecompanion.lua:11>
	[C]: in function 'cmd'
	.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: in function <.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:16>
stack traceback:
	[C]: in function 'cmd'
	.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: in function <.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:16>
```

**Reply by olimorris** [reply]
Great spot and such a dumb mistake on my part. Fixed that in the latest commit

**Reply by mrjones2014** [reply]
lol, no worries, I have _definitely_ done stuff like that 😛

**Reply by olimorris** [reply]
Let me know how you're finding it!

**Reply by mrjones2014** [reply]
Awesome so far! There's a few rough edges (some I've already done PRs for 😛) but for the most part it its very awesome.

**Reply by olimorris** [reply]
Great to hear. It genuinely changed my workflow in Neovim for the better. Lot's more I'd like to add to it in time.

Oh and there's some handy keymaps to work with the chat buffer btw.
